runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java

```diff
@@ -381,8 +381,20 @@ public void onPageLoadStarted(XWalkViewInternal view, String url) {
     @XWalkAPI
     public void onPageLoadStopped(XWalkViewInternal view, String url, LoadStatusInternal status) {
     }
-
-    private boolean onJsAlert(XWalkViewInternal view, String url, String message,
+    
+    /**
+     * Tell the client to display an alert dialog to the user.
+     * WARN: Please DO NOT override this API and onJavascriptModalDialog API in the 
+     *       same subclass to avoid unexpected behavior. 
+     * @param view the owner XWalkViewInternal instance.
+     * @param url the url of the web page which wants to show this dialog.
+     * @param message the message to be shown.
+     * @param result the callback to handle the result from caller.
+     * @return boolean Whether the client will handle the confirm dialog.
```

Comment:
Sorry, I missed one thing. There is no need to write the type of return value. `@return Wether ...` is okay.